File: Level1 - 키패드 누르기.py
def findDistance(currentN, nextN):
    keypad = {1: [0, 0], 2: [1, 0], 3: [2, 0],
              4: [0, 1], 5: [1, 1], 6: [2, 1],
              7: [0, 2], 8: [1, 2], 9: [2, 2],
              "*": [0, 3], 0: [1, 3], "#": [2, 3], }

    x1, y1 = keypad[currentN]
    x2, y2 = keypad[nextN]

    return abs(x1 - x2) + abs(y1 - y2)


def solution(numbers, hand):
    answer = ''
    current_l = '*'
    current_r = '#'

    for i in numbers:
        if i in [1, 4, 7]:
            answer += 'L'
            current_l = i
        elif i in [3, 6, 9]:
            answer += 'R'
            current_r = i
        elif i in [2, 5, 8, 0]:
            if findDistance(current_l, i) > findDistance(current_r, i):
                answer += 'R'
                current_r = i
            elif findDistance(current_l, i) < findDistance(current_r, i):
                answer += 'L'
                current_l = i
            else:
                if hand == 'left':
                    answer += 'L'
                    current_l = i
                else:
                    answer += 'R'
                    current_r = i

    return answer

File: Level1 - 키패드 누르기.js
function solution(numbers, hand) {
  const keypad = {
    1: [0, 0],
    2: [0, 1],
    3: [0, 2],
    4: [1, 0],
    5: [1, 1],
    6: [1, 2],
    7: [2, 0],
    8: [2, 1],
    9: [2, 2],
    '*': [3, 0],
    0: [3, 1],
    '#': [3, 2],
  }

  let answer = '';
  let curLeft = keypad["*"];
  let curRight = keypad["#"];

  numbers.forEach(v => {
    if (v % 3 === 1 || v === '*') {
      answer += "L"
      curLeft = keypad[v];
    } else if ((v % 3 === 0 && v !== 0) || v === '#') {
      answer += "R"
      curRight = keypad[v];
    } else {
      let left = Distance(curLeft, keypad[v]);
      let right = Distance(curRight, keypad[v]);
      if (left < right) {
        answer += "L";
        curLeft = keypad[v];
      } else if (left > right) {
        answer += "R";
        curRight = keypad[v];
      } else {
        if (hand === "left") {
          answer += "L"
          curLeft = keypad[v];
        } else {
          answer += "R";
          curRight = keypad[v];
        }
      }
    }
  });

  return answer;
}

function Distance(arr1, arr2) {
  let d = Math.abs(arr1[0] - arr2[0]) + Math.abs(arr1[1] - arr2[1]);
  return d;
}